test/metabase/api/activity_test.clj

```diff
@@ -34,45 +40,39 @@
                                     :archived    true
                                     :creator_id  (mt/user->id :crowberto)}
                  Table     table-1 {:name "rand-name"}]
-    (mt/with-model-cleanup [ViewLog]
+    (mt/with-model-cleanup [:model/RecentViews]
       (mt/with-test-user :crowberto
-        (mt/with-temporary-setting-values [user-recent-views []]
-          (doseq [{:keys [topic item]} [{:topic :event/dashboard-read :item dash-1}
-                                        {:topic :event/dashboard-read :item dash-2}
-                                        {:topic :event/dashboard-read :item dash-3}
-                                        {:topic :event/card-query :item card-1}
-                                        {:topic :event/table-read :item table-1}]]
-            (events/publish-event!
-             topic
-             ;; view log entries look for the `:actor_id` in the item being viewed to set that view's :user_id
-             (assoc item :actor_id (mt/user->id :crowberto))))
-          (testing "most_recently_viewed_dashboard endpoint shows the current user's most recently viewed dashboard."
-            (is (= dash-3 #_dash-2 ;; TODO: this should be dash-2, because dash-3 is archived
-                   (mt/user-http-request :crowberto :get 200 "activity/most_recently_viewed_dashboard"))))))
+        (doseq [{:keys [topic item]} [{:topic :event/dashboard-read :item dash-1}
+                                      {:topic :event/dashboard-read :item dash-2}
+                                      {:topic :event/dashboard-read :item dash-3}
+                                      {:topic :event/card-query :item {:card_id (u/the-id card-1)}}
+                                      {:topic :event/table-read :item table-1}]]
+          (events/publish-event! topic item))
+        (testing "most_recently_viewed_dashboard endpoint shows the current user's most recently viewed dashboard."
+          (is (= dash-3 #_dash-2 ;; TODO: this should be dash-2, because dash-3 is archived
```

Comment:
Is this a bug in the recent views query that's still being worked on, or is there something else needed for this TODO?

Reply:
This was there before, seems like a existing bug with the previous query. I'll add it to my todo list to fix but I don't think it needs to block this PR